src/pkg/artifact/model/model.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"time"
+
+	"github.com/goharbor/harbor/src/common/models"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// Artifact is the abstract object managed by Harbor. It hides the
+// underlying concrete detail and provides an unified artifact view
+// for all users.
+type Artifact struct {
+	ID                      int64
+	Type                    string // image, chart, etc
+	Repository              *models.RepoRecord
+	Tags                    []*Tag // the list of tags that attached to the artifact
+	MediaType               string // the specific media type
```

Comment:
Still thinking the positions of `Type` and `MediaType`. `Type` is generated based on `MediaType`, so maybe no need to store both two, just a type function is ok to get `Type` from `MediaType`.

Reply:
For some kinds of artifacts, such as helm chart and single manifest image, the type can be generated by the `manifest.config.mediatype`, but for index there is still no stardard to tell us that what the artifact is. So the media type here is the media type of top level manifest, not the config's. Just in case we need it

Reply:
Add both `MediaType` and `ManifestMediaType`